apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -588,29 +597,43 @@ export async function fetchRetry(
   try {
     const response = await fetch(input, opts);
 
-    switch (response.status) {
-      // 422 Unprocessable Entity, User not found in course or is not a student
-      case 422:
-        return response;
-    }
-
     if (!response.ok) {
-      throw new AugmentedError('LTI 1.3 fetch error, please try again', {
+      const resObject = (await response.json()) ?? {};
+      /*
+      Canvas error example with nested objects and message property.
+      {
+        errors: {
+          type: 'unprocessable_entity',
+          message: 'This course has concluded. AGS requests will no longer be accepted for this course.'
+        }
+      }
+      */
+      const msg = findValueByKey(resObject, 'message');
```

Comment:
This should probably handle the case where `message` isn't found, e.g. when dealing with non-Canvas systems (or if Canvas is even less consistent than we thought, which wouldn't surprise me). Of course, we'll probably want to do manual testing with new LMSes as we add them, and maybe hardcode different behavior in those cases. Maybe we just show the entire stringified response in that case? Or just the `statusText`?